lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb

```diff
@@ -0,0 +1,201 @@
+require 'fileutils'
+
+Puppet::Type.type(:bootstrap_rpm).provide(:bootstrap_rpm) do
+
+  commands :rpmbuild => 'rpmbuild'
+  commands :rpm => 'rpm'
+  commands :rpm2cpio => 'rpm2cpio'
+  commands :cpio => 'cpio'
+
+  def create
+    copy_sources
+    write_specfile
+    build_rpm
+
+    if rpm_changed?
+      copy_rpm
+      copy_srpm
+      link_rpm
+    end
+  ensure
+    FileUtils.remove_dir(base_dir) if File.directory?(base_dir)
+    @base_dir = nil
+  end
+
+  def exists?
+    copy_sources
+    write_specfile
+    build_rpm
```

Comment:
Building an RPM on every single run feels a bit heavy. Isn't there a way to avoid this? For example, you can add a `Provides: katello-ca-cert(hash) = $HASH` statement and similar for the script. That way you can query it with `rpm`